jdrfandroidbleparser/src/test/java/org/ehealthinnovation/jdrfandroidbleparser/BaseTest.kt

```diff
@@ -37,6 +37,9 @@ open class BaseTest {
                     FORMAT_SINT8, FORMAT_UINT8, FORMAT_SINT16, FORMAT_UINT16 -> {
                         val offset: Int = it.getArgument(1)
                         FormatType.fromType(formatType)?.let {
+                            if( payload.size <= offset+it.length()-1){
+                                throw IllegalArgumentException("Trying to access non-existing byte")
+                            }
```

Comment:
Why are you adding this? if you try to access outside the range it will throw an IndexOutOfRangeException?

Reply:
yes. Currently, the mock allows accessing two data bytes even there is only one byte in the payload.

Reply:
Let's talk about this today.